lib/ansible/modules/cloud/openstack/os_volume_snapshot.py

```diff
@@ -0,0 +1,178 @@
+EXAMPLES = '''
+# Creates a new volume snapshot
+- name: create a volume
+  hosts: localhost
+  tasks:
+  - name: create snapshot from volume
+    os_volume_snapshot:
+      state: present
+      cloud: default
+      name: my_volume_snapshot
+      volume: my_volume
+  - name: delete snapshot
+    os_volume_snapshot:
+      state: absent
+      cloud: default
+      name: my_volume_snapshot
+'''
+
+RETURN = '''
+snapshot:
+    description: Dictionary describing the snapshot.
+    returned: On success when I(state) is 'present'.
+    type: complex
+    contains:
+        id:
+            description: Unique snapshot ID
+            type: string
+            sample: "3fa9818f-c216-4843-b7b2-4626ce0c2573"
+        name:
+            description: Snapshot name
+            type: string
+            sample: "my_volume_snapshot"
+        size:
+            description: Snapshot size
+            type: int
+            sample: 40
+        status:
+            description: Snapshot status
+            type: string
+            sample: "available"
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.openstack import openstack_full_argument_spec
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+
+def _present_volume_snapshot(module, cloud):
+    s_name = cloud.get_volume_snapshot(module.params['name'])
+    if s_name:
+        module.exit_json(changed=False, id=s_name['id'], snapshot=s_name)
+
+    v_id = cloud.get_volume_id(module.params['volume'])
+    if not v_id:
+        module.fail_json(msg='volume %s not found' % module.params['volume'])
+
+    volume_snapshot_args = dict(
+        name=module.params['name'],
+        force=module.params['force'],
+        volume_id=v_id,
+        description=module.params['description'],
+    )
+
+    snapshot = cloud.create_volume_snapshot(
+        wait=module.params['wait'], timeout=module.params['timeout'],
+        **volume_snapshot_args)
+    module.exit_json(changed=True, id=snapshot['id'], snapshot=snapshot)
+
+
+def _absent_volume_snapshot(module, cloud):
+    try:
+        cloud.delete_volume_snapshot(
+            name_or_id=module.params['name'],
+            wait=module.params['wait'],
+            timeout=module.params['timeout'])
+    except shade.OpenStackCloudTimeout:
+        module.exit_json(changed=False)
+    module.exit_json(changed=True)
+
+
+def main():
+    argument_spec = openstack_full_argument_spec(
+        name=dict(required=True),
+        description=dict(default=None),
+        force=dict(default=False),
+        volume=dict(default=None),
+        state=dict(default='present', choices=['absent', 'present']),
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+
+    state = module.params['state']
+
+    if module.check_mode:
+        module.exit_json(changed=_system_state_change(module, service))
```

Comment:
you have no function named '_system_state_change'. You need to work that in, see theother module as an example.